src/Operations.jl

```diff
@@ -975,13 +1002,18 @@ function add(ctx::Context, pkgs::Vector{PackageSpec}, new_git=UUID[]; strict::Bo
     update_manifest!(ctx, pkgs)
     # TODO is it still necessary to prune? I don't think so..
     new_apply = download_source(ctx, pkgs)
+
+    # After downloading resolutionary packages, search for Artifact.toml files
+    # and ensure they are all downloaded and unpacked as well:
+    download_artifacts(ctx, pkgs; platform=platform)
```

Comment:
Shuold this happen after writing the env (like `build`) ?

Reply:
Next on my TODO is to modify `write_env()` to include links to the Artifacts.toml files so that we can `gc` things appropriately, so no, I think it should stay where it is.